instrumentation/java-concurrent/src/main/java/io/opentelemetry/auto/instrumentation/java/concurrent/AsyncPropagatingDisableInstrumentation.java

```diff
@@ -87,18 +85,18 @@ public DisableAsyncInstrumentation(
   public static class DisableAsyncAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static AgentScope enter() {
-      final AgentSpan span = activeSpan();
-      if (span != null) {
-        return activateSpan(noopSpan(), false);
+    public static SpanScopePair enter() {
+      final Span span = TRACER.getCurrentSpan();
+      if (span.getContext().isValid()) {
+        return new SpanScopePair(span, TRACER.withSpan(span));
```

Comment:
```suggestion
        return TRACER.withSpan(DefaultSpan.getInvalid());
```

and pass `Scope` to `@OnMethodExit`

(the idea behind this instrumentation is that it prevents the current span from propagating into async work that should not be considered as part of the current request)